src/gs-shell.c

```diff
@@ -2090,19 +2131,11 @@ gs_shell_file_to_app_cb (GObject *source,
 
 	removable_media_cat = gs_app_get_metadata_item (app, "EndlessOS::RemovableMediaCategory");
 	if (removable_media_cat != NULL) {
-		GsCategory *category;
-		if (gs_app_get_metadata_item (app, "EndlessOS::ReloadOverview") == NULL) {
-			reload_overview_and_select_category (shell, removable_media_cat);
-		} else {
-			category = side_filter_get_category_by_name (shell, removable_media_cat);
-			if (category == NULL)
-				reload_overview_and_select_category (shell, removable_media_cat);
-			else {
-				gs_shell_change_mode (shell, GS_SHELL_MODE_CATEGORY,
-						      (gpointer) category, TRUE);
-			}
-		}
-		gs_shell_activate (shell);
+		gboolean reload_overview =
+			(gs_app_get_metadata_item (app, "EndlessOS::ReloadOverview") == NULL);
+		gs_shell_reload_and_select_removable_media_category (shell, reload_overview,
+								     TRUE, removable_media_cat);
```

Comment:
Again, this could be made more clear.

Reply:
This also goes away in recent simplification